core/src/main/java/bisq/core/provider/price/PriceFeedService.java

```diff
@@ -301,28 +296,16 @@ public Date getLastRequestTimeStamp() {
         return new Date(epochInMillisAtLastRequest);
     }
 
-    public void applyLatestBisqMarketPrice(Set<TradeStatistics3> tradeStatisticsSet) {
-        // takes about 10 ms for 5000 items
-        Map<String, List<TradeStatistics3>> mapByCurrencyCode = new HashMap<>();
-        tradeStatisticsSet.forEach(e -> {
-            List<TradeStatistics3> list;
-            String currencyCode = e.getCurrency();
-            if (mapByCurrencyCode.containsKey(currencyCode)) {
-                list = mapByCurrencyCode.get(currencyCode);
-            } else {
-                list = new ArrayList<>();
-                mapByCurrencyCode.put(currencyCode, list);
+    public void applyInitialBisqMarketPrice(Map<String, Price> priceByCurrencyCode) {
+        priceByCurrencyCode.forEach((currencyCode, price) -> {
+            if (!cache.containsKey(currencyCode) || !cache.get(currencyCode).isExternallyProvidedPrice()) {
+                cache.put(currencyCode, new MarketPrice(currencyCode,
```

Comment:
Those lines are duplicate to the code in setBisqMarketPrice, maybe extract it as a method.